From 5f91ed0b5345e132d9fe1fa815bfd968778df5ea Mon Sep 17 00:00:00 2001
From: Guido Trentalancia <guido@trentalancia.net>
Date: Fri, 15 Apr 2016 13:06:45 +0200
Subject: [PATCH] dracut: 10i18n: support default loadkeys/setfont data paths
 using symbolic links

Avoid keymap/font not found error when loadkeys/setfont
are compiled with the default data directory path.

Signed-off-by: Guido Trentalancia <guido@trentalancia.net>
---
 modules.d/10i18n/module-setup.sh | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/modules.d/10i18n/module-setup.sh b/modules.d/10i18n/module-setup.sh
index 9439875..9618aa7 100755
--- a/modules.d/10i18n/module-setup.sh
+++ b/modules.d/10i18n/module-setup.sh
@@ -100,6 +100,13 @@ install() {
             inst_rules ${moddir}/10-console.rules
             inst_hook cmdline 20 "${moddir}/parse-i18n.sh"
         fi
+
+        if [[ ${kbddir} != "/usr/share" ]]; then
+            inst_dir /usr/share
+            for _src in $(eval echo {${KBDSUBDIRS}}); do
+                [ ! -e "${initdir}/usr/share/${_src}" ] && ln -s "${kbddir}/${_src}" "${initdir}/usr/share/${_src}"
+            done
+        fi
     }
 
     install_all_kbd() {
